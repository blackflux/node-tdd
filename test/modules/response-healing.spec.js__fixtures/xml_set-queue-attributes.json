{
  "fn": "call",
  "params": [
    "sqs:SetQueueAttributesCommand",
    {
      "QueueUrl": "$queueUrl",
      "Attributes": {
        "RedrivePolicy": "{\"deadLetterTargetArn\":\"arn:aws:sqs:us-west-2:12345678910:some-queue-arn\",\"maxReceiveCount\":10}"
      }
    }
  ],
  "cassette": [
    {
      "scope": "https://sqs.us-west-2.amazonaws.com:443",
      "method": "POST",
      "path": "/",
      "body": "QueueUrl=https%3A%2F%2Fsqs.us-west-2.amazonaws.com%2F12345678910%2Fsome-queue-url&Attribute.1.Name=RedrivePolicy&Attribute.1.Value=%7B%22deadLetterTargetArn%22%3A%22arn%3Aaws%3Asqs%3Aus-west-2%3A799715345196%3Asome-queue-arn%22%2C%22maxReceiveCount%22%3A10%7D&Action=SetQueueAttributes&Version=2012-11-05",
      "status": 200,
      "response": "<?xml version=\"1.0\"?><SetQueueAttributesResponse xmlns=\"http://queue.amazonaws.com/doc/2012-11-05/\"><ResponseMetadata><RequestId>f6b2708b-5902-5f1c-b06d-2bbc8a19eb95</RequestId></ResponseMetadata></SetQueueAttributesResponse>",
      "responseIsBinary": false
    }  ]
}
