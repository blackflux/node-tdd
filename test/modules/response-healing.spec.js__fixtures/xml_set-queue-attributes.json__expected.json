[
  {
    "scope": "https://sqs.us-west-2.amazonaws.com:443",
    "method": "POST",
    "path": "/",
    "body": {
      "Attributes": {
        "RedrivePolicy": "{\"deadLetterTargetArn\":\"arn:aws:sqs:us-west-2:12345678910:some-queue-arn\",\"maxReceiveCount\":10}"
      },
      "QueueUrl": "https://sqs.us-west-2.amazonaws.com/123456789101/service-name-data-local-SomeQueue"
    },
    "status": 200,
    "response": {},
    "responseIsBinary": false,
    "reqheaders": {
      "amz-sdk-invocation-id": "^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}$",
      "amz-sdk-request": "attempt=1; max=3",
      "authorization": "^AWS4-HMAC-SHA256 Credential=.+$",
      "content-length": "^[1-9][0-9]*$",
      "content-type": "application/x-amz-json-1.0",
      "host": "sqs.us-west-2.amazonaws.com",
      "user-agent": "^aws-sdk-js/.+$",
      "x-amz-content-sha256": "^[a-f0-9]{64}$",
      "x-amz-date": "^[0-9]{8}T[0-9]{6}Z$",
      "x-amz-target": "AmazonSQS.SetQueueAttributes",
      "x-amz-user-agent": "^aws-sdk-js/[0-9]+.[0-9]+.[0-9]+$",
      "x-amzn-query-mode": "true"
    }
  }
]
